State[] states
---
int16 next_horizon